src/adapters/mooar-bp-1/index.ts

```diff
@@ -0,0 +1,164 @@
+import { EventData } from "web3-eth-contract";
+import { ISaleEntity } from "../../sdk/Interfaces";
+import BasicProvider, { IBasicProviderOptions } from "../../sdk/basic-provider";
+enum ItemType {
+    NATIVE = 0,
+    ERC20 = 1,
+    ERC721 = 2,
+    ERC1155 = 3,
+}
+
+type TradeItem = {
+    itemType: ItemType;
+    token: string;
+    identifier: BigNumber;
+    amount: BigNumber;
+};
+
+class Mooar extends BasicProvider {
+    defaultPaymentToken: string;
+
+    constructor(options: IBasicProviderOptions) {
+        super(options);
+
+        this.requireDefaultPaymentToken();
+        this.events = ["OrderFulfilled"];
+    }
+
+    process = async (event: EventData): Promise<ISaleEntity | undefined> => {
+        const block = await this.sdk.getBlock(event.blockNumber);
+        const timestamp = moment.unix(block.timestamp).utc();
+
+        const offerer = event.returnValues.offerer;
+        const recipient = event.returnValues.recipient;
+        const offerRaw = event.returnValues.offer;
+        const considerationRaw = event.returnValues.consideration;
+
+        const offer: TradeItem[] = this.toTradeItems(offerRaw);
+        const consideration: TradeItem[] = this.toTradeItems(considerationRaw);
+
+        if (offerer.toLowerCase() === recipient.toLowerCase()) {
+            return;
+        }
+        if (consideration.length === 0) {
+            return;
+        }
+        const offerNft = offer.find(item => item.itemType === ItemType.ERC721 || item.itemType === ItemType.ERC1155);
+        const considerationNft = consideration.find(
+            item => item.itemType === ItemType.ERC721 || item.itemType === ItemType.ERC1155,
+        );
+        if (offerNft !== undefined && considerationNft !== undefined) {
+            if (
+                offerNft.token !== considerationNft.token ||
+                offerNft.identifier.toString() !== considerationNft.identifier.toString()
+            ) {
+                return;
+            }
+        } else if (offerNft === undefined && considerationNft === undefined) {
+            return;
+        }
+        const isOfferNft = offerNft !== undefined;
+        let nft: {
+            id: string;
+            amount: number;
+            contract: string;
+        } | null;
+        let tradePrice: {
+            token: string;
+            price: BigNumber;
+        } | null;
+        if (isOfferNft) {
+            nft = this.getNft(offer);
+            tradePrice = this.getTradePrice(consideration);
+        } else {
+            nft = this.getNft(consideration);
+            tradePrice = this.getTradePrice(offer);
+        }
+        if (nft === null || tradePrice === null) {
+            return;
+        }
+        const entity = {
+            providerName: this.name,
+            providerContract: this.contract,
+            chainId: this.sdk.chainId,
+            nfts: [nft],
+            token: tradePrice.token,
+            price: tradePrice.price,
```

Comment:
Price always has to be in WEI. I've noticed, that for native tokens (e.g. ETH, MATIC) prices are in normal values (e.g. `0.05` instead of `50000000000000000`).

Example transaction for ETH - `0x7c417970ed72cf8eb7d2beb590ddc6b135bea7a72dc9620a1161b82363d455e3`
Example transaction for Polygon - `0xbfa768cbd529a483b8a718ab202776c1dcd69c4a19c1713d526b40b06c24ad26`

Reply:
`tradePrice.price`  is `BigNumber` type here, it actually to be in WEI for native token and ERC20 token.